zebrad/src/components/mempool/crawler.rs

```diff
@@ -106,6 +119,14 @@ where
             transaction_ids.len()
         );
 
-        // TODO: Send transaction IDs to the download and verify stream (#2650)
+        if !transaction_ids.is_empty() {
+            self.mempool
+                .ready_and()
+                .await?
+                .call(mempool::Request::Queue(transaction_ids))
+                .await?;
```

Comment:
If we get an error from the mempool request, the crawler task will silently exit.
That seems like it could be confusing.

Currently, the mempool never returns an error.

But if it did, it would mean that the peer had done something wrong or unexpected:
https://github.com/ZcashFoundation/zebra/pull/2764/files#diff-ec634b649fdc36d20e3746cb9106eefafff1315ce871dd9b16fbbe5708d85784R267-R270

So do we just want to log response errors here?
(The service readiness error handling is ok, we only expect it to happen on shutdown.)

Reply:
We might also want to have tests for both the service readiness and response errors.

Reply:
Done, I think :thinking: At least I added a test to see if the crawler survives download and service errors: (9cb7499)[https://github.com/ZcashFoundation/zebra/pull/2801/commits/9cb74991debad7da734e759363c1128f3ea754ec].

Reply:
Thanks for the comprehensive tests!